core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -575,18 +537,39 @@ public Integer getMappedPort(final int originalPort) {
         }
     }
 
+    /**
+     * <b>Resolve</b> Docker image and set it.
+     *
+     * @param dockerImageName image name
+     */
+    @SneakyThrows
     public void setDockerImageName(@NonNull String dockerImageName) {
         DockerImageName.validate(dockerImageName);
-        this.dockerImageName = dockerImageName;
 
-        Profiler profiler = new Profiler("Rule creation - prefetch image");
-        profiler.setLogger(logger());
+        String[] parts = dockerImageName.split(":");
+        RemoteDockerImage remoteDockerImage = new RemoteDockerImage(parts[0], parts[1]);
         try {
-            pullImageIfNeeded(dockerImageName, profiler);
-        } catch (InterruptedException e) {
-            throw new ContainerFetchException("Failed to fetch container image for " + dockerImageName, e);
-        } finally {
-            profiler.stop().log();
+            // Mimic old behavior where we resolve image once it's set
+            remoteDockerImage.get();
```

Comment:
not sure if we should keep it or not - currently it's here for backward compatibility, but shouldn't break anything if we will remove it (and move image resolve to the "start" container's phase)